# Task 1

### Subtask a.

In [36]:
import numpy as np
from fractions import Fraction as f
from numpy import linalg as la
np.set_printoptions(precision=3,suppress=True)

In [39]:
#def f(p,q=1):
#    return p/q

#A=np.array([[f(0),f(1,2),f(0),f(0)],[f(1,3),f(0),f(0),f(0)],[f(1,3),f(1,2),f(0),f(0)],[f(1,3),f(0),f(1,1),f(1)]])
A=np.array([[0,1/2,0,0],[1/3,0,0,0],[1/3,1/2,0,0],[1/3,0,1/1,1]])
print(A)
#la.det(A)
x_0=np.array([0,1,0,0])

A_n=A
for i in range(21):
    A_n = np.dot(A_n,A)
x_n = np.dot(A_n,x_0)
x_n

[[0.    0.5   0.    0.   ]
 [0.333 0.    0.    0.   ]
 [0.333 0.5   0.    0.   ]
 [0.333 0.    1.    1.   ]]


array([0., 0., 0., 1.])

In [31]:
la.eig(A)

(array([ 1.        ,  0.        ,  0.40824829, -0.40824829]),
 array([[ 0.        ,  0.        ,  0.23462955,  0.73820659],
        [ 0.        ,  0.        ,  0.19157423, -0.60274316],
        [ 0.        ,  0.70710678,  0.42620378,  0.13546343],
        [ 1.        , -0.70710678, -0.85240755, -0.27092687]]))

### Subtask c.

Since every column of $A_1$ sums to 1, every column of $0.85A_1$
  sums to $0.85$, and since every column of $A_2$ sums to 1, every column of
  $0.15A_2$ sums to $0.15$, so (finally) every column of $0.85A_1 +
  0.15A_2$ sums to 1.